src/spuro.h

```diff
@@ -153,24 +153,24 @@ void spr_logf_(const Spuro spr, SpuroLevel level, SpuroColor color, SpuroLoc loc
 
 // Utility log macros
 
-#define spr_logf(spr, format, ...) spr_logf_((spr), SPR_NOLVL, SPR_COLOR_AUTO, SPR_HERE, false, false, (format), ## __VA_ARGS__)
-#define spr_tlogf(spr, format, ...) spr_logf_((spr), SPR_NOLVL, SPR_COLOR_AUTO, SPR_HERE, false, true, (format), ## __VA_ARGS__)
-#define spr_clogf(spr, color, format, ...) spr_logf_((spr), SPR_NOLVL, (color), SPR_HERE, false, false, (format), ## __VA_ARGS__)
-#define spr_tclogf(spr, color, format, ...) spr_logf_((spr), SPR_NOLVL, (color), SPR_HERE, false, true, (format), ## __VA_ARGS__)
-#define spr_logf_to(spr, level, format, ...) spr_logf_((spr), (level), SPR_COLOR_AUTO, SPR_HERE, false, false, (format), ## __VA_ARGS__)
-#define spr_tlogf_to(spr, level, format, ...) spr_logf_((spr), (level), SPR_COLOR_AUTO, SPR_HERE, false, true, (format), ## __VA_ARGS__)
-#define spr_clogf_to(spr, color, level, format, ...) spr_logf_((spr), (level), (color), SPR_HERE, false, false, (format), ## __VA_ARGS__)
-#define spr_tclogf_to(spr, color, level, format, ...) spr_logf_((spr), (level), (color), SPR_HERE, false, true, (format), ## __VA_ARGS__)
+#define spr_logf(spr, ...) spr_logf_((spr), SPR_NOLVL, SPR_COLOR_AUTO, SPR_HERE, false, false, __VA_ARGS__)
```

Comment:
I'm wondering how well does this interact with the zero-arg case the issue mentions.

Say, spr_logf(my_spr, "hey") versus spr_logf(my_spr, "%s", "hey")

Are they equivalent? Why the hell was I using ## in the first place then?

Reply:
> Say, spr_logf(my_spr, "hey") versus spr_logf(my_spr, "%s", "hey")
> 
> Are they equivalent? Why the hell was I using ## in the first place then?

I believe you were using ## to remove the `,` after second parameter [same is suggested in this post](https://stackoverflow.com/questions/52891546/what-does-va-args-mean)

Accepted answer on the same post starts like, "It's a non-portable syntax introduced by gcc to specifically deal with this corner case of passing no arguments at all."

Other comments on the same post also suggests that it is not available with every compiler.

Issue #3 is also using a different compiler than gcc so that's why this is now an issue.

This PR includes changes which removes the requirement to use ##
- _spr_logf_ function always needs a format argument because of function syntax.

These change makes it possible to pass format in-place of ... and __VA_ARGS__ when no other arguments are supplied.

So, spr_logf(my_spr, "hey") and spr_logf(my_spr, "%s", "hey") both will translate to same thing.

```
spr_logf_(my_spr, SPR_NOLVL, SPR_COLOR_AUTO, SPR_HERE, false, false, "hey")
```

Without these changes, clang was seeing trailing comma because ## is supported by gcc and not clang

Reply:
> I believe you were using ## to remove the `,` after second parameter [same is suggested in this post](https://stackoverflow.com/questions/52891546/what-does-va-args-mean)
> These change makes it possible to pass format in-place of ... and **VA_ARGS** when no other arguments are supplied.
> 
> So, spr_logf(my_spr, "hey") and spr_logf(my_spr, "%s", "hey") both will translate to same thing.
> Without these changes, clang was seeing trailing comma because ## is supported by gcc and not clang

So these changes don't have any drawback? The `##` was not needed?

I feel like I'm missing some problem that could arise. Maybe there's nothing.
Thanks for your patience.